File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] result = new int[temperatures.length];
        Stack<Integer> stack = new Stack<>();

        for(int idx = 0; idx < temperatures.length; idx++){
            while(!stack.isEmpty() && temperatures[stack.peek()] < temperatures[idx]){
                result[stack.peek()] = idx - stack.pop();
            }
            stack.push(idx);
        }
        return result;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        temprature = [0] * len(temperatures)
        for idx, value in enumerate(temperatures):
            while stack and stack[-1][1] < value:
                    index, _ = stack.pop()
                    temprature[index] = idx - index
            stack.append((idx, value))
        return temprature